records/generator.go

```diff
@@ -494,3 +588,29 @@ func getProto(pair string) (string, string, error) {
 	}
 	return h[0], h[1], nil
 }
+
+func specEachLabel(s string, spec labels.Func) string {
+	labels := strings.Split(s, ".")
+	specLabels := make([]string, 0, len(labels))
+	for _, l := range labels {
+		specLabels = append(specLabels, spec(l))
+	}
+	return strings.Join(specLabels, ".")
+}
+
+// return the slave number from a Mesos slave id
+func slaveIDTail(slaveID string) string {
+	fields := strings.Split(slaveID, "-")
+	return strings.ToLower(fields[len(fields)-1])
+}
+
+// BUG: The probability of hashing collisions is too high with only 17 bits.
+// NOTE: Using a numerical base as high as valid characters in DNS names would
+// reduce the resulting length without risking more collisions.
+func hashString(s string) string {
```

Comment:
It's pretty long then. Using a hash is fine, but the collision probability should be low for the cluster sizes we target. Compare generalized birthday paradox.